docs/encyclopedia/nexus-operations.mdx

```diff
@@ -167,6 +170,78 @@ This differs from how Activities and Workflows handle errors and retries:
 
 See [errors in Nexus handlers](/nexus/error-handling#errors-in-nexus-handlers) to control the retry behavior by returning a [non-retryable Nexus error](/references/failures#non-retryable-nexus-errors).
 
+## Circuit breaking {#circuit-breaking}
+
+Nexus provides a circuit breaker for each destination pair (caller Namespace and target Nexus Endpoint). The circuit breaker for each pair trips and resets independently.
+By default, the circuit breaker kicks in after 5 consecutive Nexus requests fail with a [retryable error](/references/failures#nexus-errors).
+For example, this happens if all Nexus Workers for a Nexus Endpoint are down and 5 consecutive requests fail due to request timeouts.
+
+Once a circuit breaker has tripped and is in the open state, the caller's Nexus Machinery will fail early and won't send requests to the target Nexus Endpoint.
+After 60 seconds in open state, it will change to the half-open state, which will allow only 1 request to be made.
+If the request is successful, then the circuit breaker changes its state to closed, and allows all requests to pass through.
+
+The circuit breaker state is surfaced in a caller Workflow's [Pending Nexus Operations](/nexus/execution-debugging#pending-operations) and [Pending Nexus Callbacks](/nexus/execution-debugging#pending-callbacks).
+It can be checked using the UI, the Temporal CLI, and the DescribeWorkflowExecution API.
+
+If the circuit breaker for a destination pair has been tripped, the [Pending Nexus Operation](/nexus/execution-debugging#pending-operations) for a [Nexus Operation Scheduled](/references/events#nexusoperationscheduled) event surfaces a State of Blocked and a BlockedReason.
+
+For example, in the UI:
+
+![Nexus Overview](/img/nexus/circuit-breaking.png)
+
+For example, in the CLI:
+
+```sh
+temporal workflow describe -w my-workflow-id
+
+Execution Info:
+  WorkflowId            my-workflow-id
+  ...
+
+Pending Activities: 0
+Pending Child Workflows: 0
+Pending Nexus Operations: 1
+
+  Endpoint                 my-nexus-endpoint
+  Service                  nexus-playground
+  Operation                sync-op-ok
+  OperationID
+  State                    Blocked
+  Attempt                  5
+  ScheduleToCloseTimeout   1d 0h 0m 0s
+  LastAttemptCompleteTime  56 seconds ago
+  LastAttemptFailure       {"message":"handler error (UPSTREAM_TIMEOUT): upstream timeout","cause":{"message":"upstream timeout","applicationFailureInfo":{"type":"NexusFailure"}},"applicationFailureInfo":{"type":"NexusHandlerError"}}
```

Comment:
I'd add a mention that you may not see any attempts for _your_ operation, it could be an attempt for a different operation that trips the circuit breaker for that endpoint.

Reply:
good point, fixed